modules/tinymce/src/core/test/ts/browser/api/dom/RangeExpandTest.ts

```diff
@@ -0,0 +1,91 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Unicode } from '@ephox/katamari';
+import { TinyHooks, TinySelections } from '@ephox/mcagar';
+import { assert } from 'chai';
+
+import RangeUtils from 'tinymce/core/api/dom/RangeUtils';
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.api.dom.RangeExpandTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+
+  const textEntry = 'A B ' + Unicode.nbsp + 'C D';
+
+  const makeBaseSelection = (editor: Editor, startOffset: number, endOffset: number, nodeOffset: number = 0) => {
+    const body = editor.getBody();
+    const range = editor.dom.createRng();
+    const node = body.childNodes[0].childNodes[nodeOffset];
+    range.setStart(node, startOffset);
+    range.setEnd(node, endOffset);
+    return range;
+  };
+
+  const compareRanges = (range1: Range, range2: Range) => {
+    assert.equal(range1.startContainer, range2.startContainer, 'Mismatching start container');
+    assert.equal(range1.startOffset, range2.startOffset, 'Mismatching start offset');
+    assert.equal(range1.endContainer, range2.endContainer, 'Mismatching end container');
+    assert.equal(range1.endOffset, range2.endOffset, 'Mismatching end offset');
+  };
+
+  it('TINY-9001: The content is empty', () => {
+    const editor = hook.editor();
+    editor.setContent('');
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 0 ], 0);
+    const startRange = editor.selection.getRng();
+    RangeUtils(editor).expand(startRange);
+    compareRanges(startRange, editor.selection.getRng());
+  });
+
+  it('TINY-9001: The cursor is in between two spaces', () => {
+    const editor = hook.editor();
+    editor.setContent(textEntry);
+    const startRange = makeBaseSelection(editor, 4, 4);
+    const endRange = RangeUtils(editor).expand(startRange);
+    compareRanges(startRange, endRange);
+  });
+
+  it('TINY-9001: The cursor is between a space and a word on the right', () => {
+    const editor = hook.editor();
+    editor.setContent(textEntry);
+    const startRange = makeBaseSelection(editor, 5, 5);
+    const endRange = RangeUtils(editor).expand(startRange);
+    compareRanges(makeBaseSelection(editor, 5, 6), endRange);
+  });
+
+  // TODO Disabled due to inconsistent behavior between left and right, see TINY-9029
```

Comment:
The difference seems to be one is a nbsp and one is a space. Could it be that it just doesn't handle nbsps as whitespace? I guess it shouldn't expand in that case.

Reply:
No, the nbsp doesn't affect it, as the check is for whitespace, not a normal space. As far as I could tell it was an off-by-one error.